packages/xod-project/src/patch.js

```diff
@@ -830,7 +830,9 @@ const sendDeferNodesToBottom = def(
  */
 export const getTopology = def(
   'getTopology :: Patch -> Either Error [NodeId]',
-  patch => R.compose(R.map(sendDeferNodesToBottom(patch)), toposortGraph)(patch)
+  R.memoizeWith(JSON.stringify, patch =>
```

Comment:
If I understand correctly, it will blow RAM usage eventually. See https://github.com/ramda/ramda/blob/v0.25.0/source/memoizeWith.js the `cache` dictionary is filled up as we get slight patch variations and nothing cleans the cache then.

`JSON.stringify` is not a speed champion either.

Does the memoization in this place bring notable performance improvements?

Reply:
Using `WeakMap` to store cache could be cool.
There are probably a dozen of libraries on npm that do this🤔